### PR TITLE
add CompositionContainer.

### PR DESCRIPTION
**Pull request by vwxyzh**

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.Engine/CompositionContainer.cs [hunk 1]
```diff
@@ -59,13 +77,22 @@ public static CompositionHost GetContainer(IEnumerable<Assembly> assemblies)
 
             try
             {
-                return configuration.CreateContainer();
+                DefaultContainer = configuration.CreateContainer();
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.Engine/CompositionContainer.cs]: >DefaultContainer [](start = 16, length = 16)

Could there be thread safety issue?

Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.Engine/CompositionContainer.cs]: There is only one caller...

**Comment by hellosnow** [comment]
>         public static CompositionHost GetContainer(IEnumerable<Assembly> assemblies)

each time CompositionContainer.GetContainer is called, default container changes?

---
Refers to: src/Microsoft.DocAsCode.Build.Engine/CompositionContainer.cs:62 in 67cc47d. [](commit_id = 67cc47de1e24fa4da8dede07fea21bce12e6b9d1, deletion_comment = False)

**Comment by vwxyzh** [comment]
>         public static CompositionHost GetContainer(IEnumerable<Assembly> assemblies)

yes, but there is one call per "build" actually.

---
In reply to: [297890563](https://github.com/dotnet/docfx/pull/1611#issuecomment-297890563) [](ancestors = 297890563)

---
Refers to: src/Microsoft.DocAsCode.Build.Engine/CompositionUtility.cs:62 in 67cc47d. [](commit_id = 67cc47de1e24fa4da8dede07fea21bce12e6b9d1, deletion_comment = False)

**Comment by vicancy** [comment]
:shipit:

**Comment by superyyrrzz** [comment]
:shipit:

**Comment by vwxyzh** [comment]
#925

**Pull request merged**